src/test/java/uk/gov/pay/ledger/report/dao/PerformanceReportDaoIT.java

```diff
@@ -106,4 +115,77 @@ public void report_volume_total_amount_and_average_amount_for_all_payments() {
         assertThat(performanceReportEntity.getTotalAmount(), is(closeTo(new BigDecimal(3000L), ZERO)));
         assertThat(performanceReportEntity.getAverageAmount(), is(closeTo(new BigDecimal(1000L), ZERO)));
     }
+
+    @Test
+    public void verifyMonthlyGatewayPerformanceReportTest() {
+        aTransactionFixture()
+                .withGatewayAccountId("1")
+                .withAmount(1000L)
+                .withState(TransactionState.STARTED)
+                .withTransactionType("PAYMENT")
+                .withLive(true)
+                .withCreatedDate(ZonedDateTime.parse("2019-01-01T02:00:00Z"))
+                .insert(rule.getJdbi());
+
+        aTransactionFixture()
+                .withGatewayAccountId("1")
+                .withAmount(1000L)
+                .withState(TransactionState.SUCCESS)
+                .withTransactionType("PAYMENT")
+                .withLive(true)
+                .withCreatedDate(ZonedDateTime.parse("2019-01-01T02:00:00Z"))
+                .insert(rule.getJdbi());
+
+        aTransactionFixture()
+                .withGatewayAccountId("1")
+                .withAmount(1000L)
+                .withState(TransactionState.SUCCESS)
+                .withTransactionType("REFUND")
+                .withLive(true)
+                .withCreatedDate(ZonedDateTime.parse("2019-01-01T02:00:00Z"))
+                .insert(rule.getJdbi());
+
+        aTransactionFixture()
+                .withGatewayAccountId("2")
+                .withAmount(1000L)
+                .withState(TransactionState.SUCCESS)
+                .withTransactionType("PAYMENT")
+                .withLive(true)
+                .withCreatedDate(ZonedDateTime.parse("2019-01-01T02:00:00Z"))
+                .insert(rule.getJdbi());
+
+        aTransactionFixture()
+                .withGatewayAccountId("2")
+                .withAmount(1000L)
+                .withState(TransactionState.SUCCESS)
+                .withTransactionType("PAYMENT")
+                .withLive(true)
+                .withCreatedDate(ZonedDateTime.parse("2019-02-01T02:00:00Z"))
+                .insert(rule.getJdbi());
```

Comment:
I think the readability of adding all the transactions above could be improved. If you look at the `report_volume_total_amount_and_average_amount_by_state` for example it's defined a list of transactions called `relevantAmounts`. It's then clear the previous 3 transactions that were added shouldn't be returned in the call to the DAO. At the moment the way this is written I have to eyeball every transaction to figure out whether the test is correct.

Reply:
Done

Reply:
Done